javascript/javascript1/week2/candy.js

```diff
@@ -0,0 +1,42 @@
+let boughtCandy = [];
+function addCandy(candyType, weight) {
+    switch(candyType) {
+        case "sweet":
+            boughtCandy.push(weight * 0.5);
+            break;
+        case "chocolate":
+            boughtCandy.push(weight * 0.7);
+            break;
+        case "toffee":
+            boughtCandy.push(weight * 1.1);
+            break;
+        case "chewing-gum":
+            boughtCandy.push(weight * 0.03);
+            break;
+    }
+}
+let amountToSpend = Math.random() * 100;
+console.log(`you can spend ${amountToSpend} on candies`)
+
+addCandy("sweet", 50);
+canBuyMoreCandy();
```

Comment:
Usually you should not try and run a function before it has been defined. `canBuyMoreCandy` is not defined until later in the file.

Reply:
Got it